public/sw.js

```diff
@@ -0,0 +1,65 @@
+importScripts('/assets/sw-toolbox/sw-toolbox.js');
+
+function networkOnlyError(request, values, options) {
+  return toolbox.networkOnly(request, values, options).catch((err) => {
+    return toolbox.cacheOnly(new Request('/offline'));
+  });
+}
+
+function networkFirstError(request, values, options) {
+  return toolbox.networkFirst(request, values, options).catch((err) => {
+    return toolbox.cacheOnly(new Request('/offline'));
+  });
+}
+
+function networkFirstIndex(request, values, options) {
+  return toolbox.networkFirst(request, values, options).catch((err) => {
+    return toolbox.cacheOnly(new Request('/'));
+  });
+}
+
+// pre cache all urls that are needed for the main and offline page
+var preCache = [
```

Comment:
My only problem here and below is maintainability, is there a way we can make this more coherent so we don't need to update this script every time we add a new route? You can make a Play template that can gets the route `documentation` from the `Router` and maybe just keep a list of excludes here. It might seem overcomplicated now but it would be a great feature to have for future-proofing.